Docs: Update "Install OpenVINO Runtime on Linux from Archive File" page (#13345)
openvinotoolkit/openvino: docs/install_guides/installing-openvino-from-archive-linux.md

# Install OpenVINO™ Runtime on Linux from an Archive File {#openvino_docs_install_guides_installing_openvino_from_archive_linux}

> With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as sample code for running demos. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the Release Notes for more information on updates in the 2022.2 release.

Reviewer: why do we remove the link to Release Notes here?

Author: Oops! There should still be a link there. I'll go back and add the link.

> With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as sample code for running demos. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the [Release Notes](https://software.intel.com/en-us/articles/OpenVINO-RelNotes) for more information on updates in the 2022.2 release.

Reviewer: ```suggestion
With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as code samples. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the [Release Notes](https://software.intel.com/en-us/articles/OpenVINO-RelNotes) for more information on updates in the 2022.2 release.
```

@sphinxdirective
.. tab:: Operating Systems

  * Ubuntu 18.04 long-term support (LTS) x86, 64-bit

>   * Ubuntu 20.04 long-term support (LTS), 64-bit

Reviewer: ```suggestion
  * Ubuntu 20.04 long-term support (LTS) x86, 64-bit
```

>   * Red Hat Enterprise Linux 8, 64-bit

Reviewer: ```suggestion
  * Red Hat Enterprise Linux 8 x86, 64-bit
```

>   * Ubuntu 18.04 long-term support (LTS), 64-bit

Reviewer: ```suggestion
  * Ubuntu 18.04 long-term support (LTS) x86, 64-bit
```

  * Ubuntu 20.04 long-term support (LTS), 64-bit

>   * Red Hat Enterprise Linux 8, 64-bit

Reviewer 2: do we have updates for [the instruction](https://docs.openvino.ai/latest/openvino_docs_install_guides_installing_openvino_raspbian.html) how to install https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_debian9_arm_2022.2.0.7713.af16ea1d79a_armhf.tgz ? It's a package for ARM arch build on Debian 9, it provides MYRIAD plugin

>   * Red Hat Enterprise Linux 8, 64-bit

Reviewer: No we don't. Is there a jira ticket for it?

>   * Red Hat Enterprise Linux 8, 64-bit

Reviewer 2: Suppose no. I just see it's a remaining document about installation which is not covered..

>   * Red Hat Enterprise Linux 8, 64-bit

Reviewer: Yes, let's create a ticket for it and update it in a separate PR. We need some specific information on how to update it. :)

>   * Red Hat Enterprise Linux 8, 64-bit

Reviewer: https://jira.devtools.intel.com/browse/CVS-95392  has been created.

  * `CMake 3.13 or higher, 64-bit <https://cmake.org/download/>`_
  * GCC 7.5.0 (for Ubuntu 18.04) or GCC 9.3.0 (for Ubuntu 20.04)
  * `Python 3.6 - 3.9, 64-bit <https://www.python.org/downloads/windows/>`_
     * Note that OpenVINO is gradually phasing out support for Python 3.6. Python 3.7 - 3.9 are recommended. 

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install the OpenVINO Core Components

First, open a terminal using Ctrl + Alt + T. Create a folder for OpenVINO and move into it by issuing the following commands. If the `/opt/intel` folder already exists, skip the `mkdir` command.

```sh
sudo mkdir /opt/intel
cd /opt/intel
```

> > **NOTE**: The `/opt/intel` path is the recommended folder path for administrators or root users. If you prefer to install OpenVINO in regular userspace, the recommended path is `/home/<USER>/Intel`. You may use a different path if desired.

Reviewer 3: ```suggestion
> **NOTE**: The `/opt/intel` path is the recommended folder path for administrators or root users. If you prefer to install OpenVINO in regular userspace, the recommended path is `/home/<USER>/intel`. You may use a different path if desired.
```
For consistency with the installer, in 2022.1 the path in $HOME was with a lowercase i.

   
      sudo wget https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_debian9_arm_2022.2.0.7713.af16ea1d79a_armhf.tgz -O openvino_2022.2.0.7713.tgz
      sudo tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_debian9_arm_2022.2.0.7713.af16ea1d79a_armhf openvino_2022.2.0.7713
     
@endsphinxdirective

Finally, create a symbolic link to the folder by issuing:

```
sudo ln -s openvino_2022.2.0.7713 openvino_2022
```
> **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

Congratulations, you finished installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO™. When other pages in OpenVINO™ documentation refer to the `<INSTALL_DIR>` directory, this is the folder they're referring to. If you installed OpenVINO™ in a different location (such as `/home/<USER>/Intel/openvino_2022`), make sure to use that instead.

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

> You must update several environment variables before you can compile and run OpenVINO™ applications. Open a terminal (if it isn't already open) and run the setupvars.bat batch file as shown below to temporarily set your environment variables. Again, if you installed OpenVINO™ in a folder other than `/opt/intel/openvino_2022`, use that location instead.

Reviewer 3: ```suggestion
You must update several environment variables before you can compile and run OpenVINO™ applications. Open a terminal (if it isn't already open) and run the setupvars.sh script as shown below to temporarily set your environment variables. Again, if you installed OpenVINO™ in a folder other than `/opt/intel/openvino_2022`, use that location instead.
```

```sh
source /opt/intel/openvino_2022/setupvars.sh
```  

If you have more than one OpenVINO™ version on your machine, you can easily switch its version by sourcing `setupvars.sh` of your choice.

> > **NOTE**: The above command must be re-run every time the terminal is opened. To set up Linux so it automatically runs the command every time a new terminal is opened, open `~/.bashrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `.bashrc` is not recommended when you have many OpenVINO™ versions on your machine and want to switch among them, as each may require different setup.

Reviewer 3: ```suggestion
> **NOTE**: The above command must be re-run every time the terminal is opened. To set up Linux so it automatically runs the command every time a new terminal is opened, open `~/.bashrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `.bashrc` is not recommended when you have multiple OpenVINO™ versions on your machine and want to switch among them, as each may require different setup.
```
Or several?

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install the OpenVINO Core Components

> 1. Open a terminal using Ctrl + Alt + T. Create a folder for OpenVINO and move into it by issuing the following commands. If the `/opt/intel` folder already exists, skip the `mkdir` command.

Reviewer: ```suggestion
1. Open a terminal using Ctrl + Alt + T. 

2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.
```

   ```sh
   sudo mkdir /opt/intel
   cd /opt/intel
   ```

>    > **NOTE**: The `/opt/intel` path is the recommended folder path for administrators or root users. If you prefer to install OpenVINO in regular userspace, the recommended path is `/home/<USER>/intel`. You may use a different path if desired.

Reviewer: ```suggestion
   > **NOTE**: The `/opt/intel` path is the recommended folder path for administrators or root users. If you prefer to install OpenVINO in regular userspace, the recommended path is `/home/<USER>/intel`. You may use a different path if desired. Make sure that you use the correct path in the steps below.
```

> 2. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), unpack it, and rename the folder to `openvino_2022.2.0.7713` by issuing:

Reviewer: ```suggestion
3. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the folder to `openvino_2022.2.0.7713` by using the following commands:
```

@sphinxdirective

.. tab:: Ubuntu 18.04

   .. code-block:: sh
   
      sudo curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
      sudo tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
      
.. tab:: Ubuntu 20.04

   .. code-block:: sh
   
      sudo wget https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64.tgz -O openvino_2022.2.0.7713.tgz
      sudo tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
      
.. tab:: Red Hat

   .. code-block:: sh
   
      sudo wget https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_rhel8_2022.2.0.7713.af16ea1d79a_x86_64.tgz -O openvino_2022.2.0.7713.tgz
      sudo tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_rhel8_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
      
@endsphinxdirective

> 3. Create a symbolic link to the folder by issuing:

Reviewer: ```suggestion
4. For simplicity, it is useful to create a symbolic link:
```

   ```
   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```

>    > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, then re-issue the previous command.

Reviewer: ```suggestion
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, and then re-run the command above.
```

Congratulations, you finished installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO™. When other pages in OpenVINO™ documentation refer to the `<INSTALL_DIR>` directory, this is the folder they're referring to. If you installed OpenVINO™ in a different location (such as `/home/<USER>/Intel/openvino_2022`), make sure to use that instead.

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

> You must update several environment variables before you can compile and run OpenVINO™ applications. Open a terminal (if it isn't already open) and run the setupvars.sh script as shown below to temporarily set your environment variables. Again, if you installed OpenVINO™ in a folder other than `/opt/intel/openvino_2022`, use that location instead.

Reviewer: ```suggestion
You must update several environment variables before you can compile and run OpenVINO applications. Open a terminal window and run the `setupvars.sh` script as shown below to temporarily set your environment variables. If your <INSTALL_DIR> is not `/opt/intel/openvino_2022`, use the correct one instead.
```

```sh
source /opt/intel/openvino_2022/setupvars.sh
```  

If you have more than one OpenVINO™ version on your machine, you can easily switch its version by sourcing `setupvars.sh` of your choice.

> > **NOTE**: The above command must be re-run every time the terminal is opened. To set up Linux so it automatically runs the command every time a new terminal is opened, open `~/.bashrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `.bashrc` is not recommended when you have multiple OpenVINO™ versions on your machine and want to switch among them.

Reviewer: ```suggestion
> **NOTE**: The above command must be re-run every time you start a new terminal session. To set up Linux to automatically run the command every time a new terminal is opened, open `~/.bashrc` in your favorite editor and add `source /opt/intel/openvino_2022/setupvars.sh` after the last line. Next time when you open a terminal, you will see `[setupvars.sh] OpenVINO™ environment initialized`. Changing `.bashrc` is not recommended when you have multiple OpenVINO versions on your machine and want to switch among them.
```

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install the OpenVINO Core Components

1. Open a command prompt terminal window. You can use the keyboard shortcut: Ctrl+Alt+T

2. Create the `/opt/intel` folder for OpenVINO and switch to it by using the following commands. If the folder already exists, skip the `mkdir` command.

   ```sh
   sudo mkdir /opt/intel
   cd /opt/intel
   ```

   > **NOTE**: The `/opt/intel` path is the recommended folder path for administrators or root users. If you prefer to install OpenVINO in regular userspace, the recommended path is `/home/<USER>/intel`. You may use a different path if desired.

> 3. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the folder to `openvino_2022.2.0.7713` by using the following commands:

Reviewer: ```suggestion
3. Download the OpenVINO Runtime archive file for Linux from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it, and rename the folder to `openvino_2022.2.0.7713` by using the following commands:
```

@sphinxdirective

.. tab:: Ubuntu 18.04

   .. code-block:: sh
   
      sudo curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
      sudo tar -xf openvino_2022.2.0.7713.tgz

>       sudo mv l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713

Reviewer: Updated by adding one more step to browse to the Downloads folder, and changing the commands. Please help check.

      
.. tab:: Ubuntu 20.04

   .. code-block:: sh
   

>       sudo wget https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64.tgz -O openvino_2022.2.0.7713.tgz

Reviewer 3: Is curl not installed by default in Ubuntu 20.04?

Reviewer: @EdjeElectronics could you help check?

Author: That was a mistake on my part! Sorry, I'll fix it 😄

Author: Okay, the Ubuntu 20.04 and Red Hat commands were changed to `curl` instead of `wget`.

      sudo tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
      
.. tab:: Red Hat

   .. code-block:: sh
   
      sudo wget https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_rhel8_2022.2.0.7713.af16ea1d79a_x86_64.tgz -O openvino_2022.2.0.7713.tgz
      sudo tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_rhel8_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
      
@endsphinxdirective

4. For simplicity, it is useful to create a symbolic link as below:

   ```
   sudo ln -s openvino_2022.2.0.7713 openvino_2022
   ```
   > **NOTE**: If you have already installed a previous release of OpenVINO 2022, a symbolic link to the `openvino_2022` folder may already exist. Remove the previous link with `sudo rm openvino_2022`, and then re-run the command above.

Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, for example, `/home/<USER>/Intel/`, OpenVINO is then installed in `/home/<USER>/Intel/openvino_2022`.

> The installation directory above is also referred as <INSTALL_DIR> throughout the documentation.

Reviewer: ```suggestion
The directory containing the `openvino_2022` folder is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.
```

> The directory containing the `openvino_2022` folder is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.

Reviewer: ```suggestion
The path to the `openvino_2022` directory is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.
```
Will this sound more clear?

Author: Yes! Good way of stating it.

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

You must update several environment variables before you can compile and run OpenVINO applications. Open a terminal window and run the `setupvars.sh` script as shown below to temporarily set your environment variables. If your <INSTALL_DIR> is not `/opt/intel/openvino_2022`, use the correct one instead.

```sh
source /opt/intel/openvino_2022/setupvars.sh
```  

> If you have more than one OpenVINO™ version on your machine, you can easily switch its version by sourcing `setupvars.sh` of your choice.

Reviewer: ```suggestion
If you have more than one OpenVINO™ version on your machine, you can easily switch its version by sourcing the `setupvars.sh` of your choice.
```

   ```sh
   sudo mkdir /opt/intel
   ```
   > **NOTE**: The `/opt/intel` path is the recommended folder path for administrators or root users. If you prefer to install OpenVINO in regular userspace, the recommended path is `/home/<USER>/intel`. You may use a different path if desired.

3. Browse to the current user's `Downloads` folder:
   ```sh
    cd <user_home>/Downloads
    ```
    
4. Download the OpenVINO Runtime archive file for Linux from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract the files, rename the extracted folder and move it to the desired path:

@sphinxdirective

.. tab:: Ubuntu 18.04

   .. code-block:: sh
   
      curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
      tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64 /opt/intel/openvino_2022.2.0.7713
      
.. tab:: Ubuntu 20.04

   .. code-block:: sh
   

>       sudo curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz

Reviewer: ```suggestion
      curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
```

      tar -xf openvino_2022.2.0.7713.tgz
      sudo mv l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64 /opt/intel/openvino_2022.2.0.7713
      

> .. tab:: Red Hat

Reviewer: ```suggestion
.. tab:: RHEL 8
```

         tar -xf openvino_2022.2.0.7713.tgz
         sudo mv l_openvino_toolkit_ubuntu18_2022.2.0.7713.af16ea1d79a_x86_64 /opt/intel/openvino_2022.2.0.7713
      
   .. tab:: Ubuntu 20.04

      .. code-block:: sh
   
         curl -L https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/linux/l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
         tar -xf openvino_2022.2.0.7713.tgz
         sudo mv l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64 /opt/intel/openvino_2022.2.0.7713
      
   .. tab:: RHEL 8

      .. code-block:: sh
   
         curl -L https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/linux/l_openvino_toolkit_rhel8_2022.2.0.7713.af16ea1d79a_x86_64.tgz --output openvino_2022.2.0.7713.tgz
         tar -xf openvino_2022.2.0.7713.tgz
         sudo mv l_openvino_toolkit_rhel8_2022.2.0.7713.af16ea1d79a_x86_64 /opt/intel/openvino_2022.2.0.7713
        
5. For simplicity, it is useful to create a symbolic link as below:
   
   .. code-block:: sh
   
      sudo ln -s openvino_2022.2.0.7713 openvino_2022
  
   .. note::

>       If you have already installed a previous release of OpenVINO 2022, a symbolic link to the ``openvino_2022`` folder may already exist. Remove the previous link with ``sudo rm openvino_2022``, and then re-run the command above.

Reviewer 2: IMO, `unlink` command is more safe than `rm`

>       If you have already installed a previous release of OpenVINO 2022, a symbolic link to the ``openvino_2022`` folder may already exist. Remove the previous link with ``sudo rm openvino_2022``, and then re-run the command above.

Reviewer: ```suggestion
      If you have already installed a previous release of OpenVINO 2022, a symbolic link to the ``openvino_2022`` folder may already exist. Unlink the previous link with ``sudo unlink openvino_2022``, and then re-run the command above.
```

Congratulations, you finished the installation! The `/opt/intel/openvino_2022` folder now contains the core components for OpenVINO. If you used a different path in Step 2, for example, `/home/<USER>/Intel/`, OpenVINO is then installed in `/home/<USER>/Intel/openvino_2022`.

> The directory containing the `openvino_2022` folder is also referred as `<INSTALL_DIR>` throughout the OpenVINO documentation.

Reviewer 3: ```suggestion
The `openvino_2022` directory is also referred to as `<INSTALL_DIR>` throughout the OpenVINO documentation.
```
The directory containing the `openvino_2022` folder is `/home<USER>/Intel` or `/opt/intel` in the example above, which is not what we refer to with INSTALL_DIR.

Reviewer: Yeah, I was going to say `<INSTALL_DIR>` includes the `openvino_2022` folder. Now I see the confusion. Thanks for pointing it out.
